File: 프로그래머스/lv2/70129. 이진 변환 반복하기/이진 변환 반복하기.java
class Solution {
    public int[] solution(String s) {

        int count = 0;
        int removed = 0;
        
        while(true){
            if(s.equals("1")){
                break;
            }else if(!s.equals("1")){
                StringBuilder binaryStr = new StringBuilder();
                for(char c:s.toCharArray()){
                    if(c == '0'){
                        removed++;
                    }else{
                        binaryStr.append(c);
                    }
                }
                s = Integer.toBinaryString(binaryStr.toString().length());
            }
            count++;
        }
        
        return new int[]{count, removed};
    }
}

File: 프로그래머스/lv2/70129. 이진 변환 반복하기/이진 변환 반복하기.py
def solution(s):
    count = 0
    removed = 0
    
    while True:
        if s == "1":
            break
        elif s != "1":
            binary_str = ""
            for c in s:
                if c == '0':
                    removed += 1
                else:
                    binary_str += c
            s = bin(len(binary_str))[2:]
        count += 1
        
    return [count, removed]